### ISSUE TITLE
"Undefined name" message about a name that isn't there in code

### ISSUE THREAD

**Issue by gnprice** [facebook/pyre-check#42]
Running Pyre on a Django app (namely Zulip), I get a lot of error messages like this:
```
zerver/apps.py:12:4 Undefined name [18]: Global name `django` is undefined.
zerver/apps.py:21:11 Undefined name [18]: Global name `django` is undefined.
zerver/apps.py:22:12 Undefined name [18]: Global name `django` is undefined.
```

OK, I see in #27 that this is a funny way of saying Pyre can't find the module `django`. But this message is completely baffling -- most of all, because the name `django` doesn't appear on any of those lines!

Instead, the lines are
```
    cache.clear()
...
        if settings.POST_MIGRATION_CACHE_FLUSHING:
            post_migrate.connect(flush_cache, sender=self)
```
The real issue, I'm guessing, is that `cache` and `settings` and `post_migrate` are all imported from various submodules of `django`.

The way mypy :P would handle this is to give an error on the actual import. If for some reason you want to only give an error if the import is actually used (but if it isn't used, wouldn't it be better to just remove the import?), it could be on the line where it's used... but the message should mention an identifier that's actually on the line it refers to.

**Comment by gnprice** [comment]
For comparison, here's what mypy reports for the same file, by default. (The normal mypy run for Zulip uses the config flag `ignore_missing_imports = True`, which suppresses exactly this sort of error.)
```
zerver/apps.py:5: error: No library stub file for module 'django.apps'
zerver/apps.py:6: error: No library stub file for module 'django.conf'
zerver/apps.py:7: error: No library stub file for module 'django.core.cache'
zerver/apps.py:8: error: No library stub file for module 'django.db.models.signals'
```

Those messages refer to the lines
```
from django.apps import AppConfig
from django.conf import settings
from django.core.cache import cache
from django.db.models.signals import post_migrate
```
and that seems a lot easier to interpret.

**Comment by sinancepel** [comment]
Hey Greg, thanks for the report - we currently error at the time of module dereference, but it's certainly worth considering.

**Comment by sinancepel** [comment]
With the following change in trunk, we now report on unresolved imports: 8e6e4cf69b054e799a986e8da3585a68cdc06968

**Comment by a-rmz** [comment]
@sinancepel is the change already in the release? Or when will it be released? I'm using version 0.0.6 and I'm getting the same error (18).

**Comment by dark** [comment]
The fix was released in `0.0.7`, uploaded to Pypi a few hours ago.

**Comment by a-rmz** [comment]
Amazing! Thanks @dark! 😄
